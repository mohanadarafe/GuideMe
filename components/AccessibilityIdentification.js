import React from 'react';
import { View, Image, StyleSheet } from 'react-native';
import { Marker } from 'react-native-maps';

function AccessibilityIdentification() {

    let handicapIconPath = "./../assets/Icons/handicap_icon.png";

    return (
        <View>

            {/* H Building  */}
            <Marker
                coordinate={{ latitude: 45.497245, longitude: -73.578968 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

            {/* LB Building */}
            <Marker
                coordinate={{ latitude: 45.496612, longitude: -73.578010 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

            {/* GM Building */}
            <Marker
                coordinate={{ latitude: 45.495899, longitude: -73.578712 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

            {/* EV Building */}
            <Marker
                coordinate={{ latitude: 45.495735, longitude: -73.578430 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

            {/* MB Building */}
            <Marker
                coordinate={{ latitude: 45.495314, longitude: -73.579055 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

            {/* SP Building  */}
            <Marker
                coordinate={{ latitude: 45.457899, longitude: -73.641310 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

            {/* CJ Building */}
            <Marker
                coordinate={{ latitude: 45.457564, longitude: -73.640394 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

            {/* CC Building  */}
            <Marker
                coordinate={{ latitude: 45.458213, longitude: -73.640150 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

            {/* AD Building */}
            <Marker
                coordinate={{ latitude: 45.458105, longitude: -73.639784 }}>
                <Image style={styles.accessibilityIdentification} source={require(handicapIconPath)} />
            </Marker>

        </View>
    );
}

export const styles = StyleSheet.create({
    accessibilityIdentification: {
        width: 30,
        height: 30,
    }
})


export { AccessibilityIdentification };