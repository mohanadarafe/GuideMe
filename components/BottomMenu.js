import React, { useEffect } from 'react';
import { View, AsyncStorage, Text, StyleSheet, Image } from "react-native";
import { ToggleCampus } from './ToggleCampus';
import { Icon, Button, Right } from 'native-base';
import { MoreDetails } from '../screens/MoreDetails';
// import { pointOfInterestInfo} from '../screens/pointOfInterestInfo';

function BottomMenu() {
    const [selectedBuilding, setSelectedBuilding] = React.useState("");
    const [iconSelected, setIconSelected] = React.useState(false);

    const buildingSelected = async () => {
        let name = await AsyncStorage.getItem('buildingSelected');
        setSelectedBuilding(name);
        
    }

    useEffect(() => {
        const intervalId = setInterval(() => {
            buildingSelected();
        }, 100)
        return () => clearInterval(intervalId);
    })

    if (iconSelected && selectedBuilding) {
        return (
            <View style={styles.moreDetails}>
                <MoreDetails name={selectedBuilding} />
                <Icon name="ios-arrow-down" style={styles.arrowDown} onPress={() => { setIconSelected(false) }} />
            </View>
        );
    }

    else if (iconSelected && !selectedBuilding) {
        return (
            <View style={styles.moreDetails}>
                {/* <pointOfInterestInfo/> */}
                <Icon name="ios-arrow-down" style={styles.arrowDown} onPress={() => { setIconSelected(false) }} />
            </View>
        );
    }

    if (!selectedBuilding) {
        return (
            <View style={styles.container}>
                <Icon name="ios-arrow-up" style={styles.arrowUp} onPress={() => { setIconSelected(true) }} />
                <Text style={styles.mainLabel}>Nearby</Text>
                <Text style={styles.shortLabel}>Food, drinks & more</Text>
                <View style={styles.toggle}>
                    <ToggleCampus />
                </View>
            </View>
        )
    }
    else {
        return (
            <View style={styles.container}>
                <Icon name="ios-arrow-up" style={styles.arrowUp} onPress={() => { setIconSelected(true) }} />
                <Text style={styles.mainLabel}>{selectedBuilding}</Text>
                <Text style={styles.shortLabel}>More info</Text>
                <View style={styles.toggle}>
                    <ToggleCampus />
                </View>
            </View>
        )
    }

}

export const styles = StyleSheet.create({
    container: {
        width: '100%',
        height: 350,
        position: 'absolute',
        borderRadius: 30.5,
        backgroundColor: '#2A2E43',
        bottom: -275
    },
    moreDetails: {
        width: '100%',
        height: '100%',
        position: 'absolute',
        backgroundColor: '#2A2E43',
        alignItems: 'center',
        justifyContent: 'space-between'
    },
    arrowUp: {
        color: '#ffffff',
        left: '5%',
        top: '7%'
    },

    toggle: {
        position: 'absolute',
        left: "80%",
        top: "6.5%"
    },
    mainLabel: {
        position: 'absolute',
        top: '5%',
        left: '12.5%',
        color: '#FFFFFF',
        fontSize: 20,
        fontFamily: 'encodeSansExpanded'
    },
    shortLabel: {
        position: 'absolute',
        top: '12%',
        left: '12.5%',
        color: '#80828D',
        fontSize: 16,
        fontFamily: 'encodeSansExpanded'
    },

    imageContainer: {
        width: '100%',
        height: '30%',
        top: '0%',
        position: 'absolute',
        backgroundColor: '#000000',
        opacity: 0.3

    },

    buildingImage: {
        width: '100%',
        height: '100%',
        top: '0%',
        position: 'relative'
    },

    arrowDown: {
        color: '#ffffff',
        top: '5%',
        fontSize: 54,
        position: 'absolute'
    },



});

export { BottomMenu };