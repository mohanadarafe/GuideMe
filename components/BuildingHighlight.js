import React from "react";
import { View, AsyncStorage } from "react-native";
import { Polygon } from "react-native-maps";
import coord from "../constants/buildingCoordinates";
import { buildingData } from "../constants/buildingData";

/**
 * US3 - As a user, I would like to be able to identify campus buildings and
 * distinguish them from other buildings.
 * The following function colors the campus buildings.
 */

function BuildingHighlight () {
    const [buildingName, setBuildingName] = React.useState("");
    AsyncStorage.setItem("buildingSelected", buildingName);
    var buildings = buildingData();
    return (
        <View>
            <Polygon
                id="tap_h"
                coordinates={coord.h.coordinates}
                tappable={true}
                onPress={() => { setBuildingName(buildings["Hall Building"].name); }}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_lb"
                coordinates={coord.lb.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["LB Building"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_gm"
                coordinates={coord.gm.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["GM Building"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_ev"
                coordinates={coord.ev.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["EV Building"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_mb"
                coordinates={coord.mb.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["JMSB"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_sp"
                coordinates={coord.sp.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["SP Building"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_cj"
                coordinates={coord.cj.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["CJ Building"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_cc"
                coordinates={coord.cc.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["CC Building"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_ad"
                coordinates={coord.ad.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["AD Building"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />

            <Polygon
                id="tap_gn"
                coordinates={coord.gn.coordinates}
                tappable={true}
                onPress={() => setBuildingName(buildings["Grey Nuns"].name)}
                fillColor="rgba(76, 79, 98, 0.7)"
            />
        </View>
    );
}

export { BuildingHighlight };