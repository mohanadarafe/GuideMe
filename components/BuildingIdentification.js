import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { Marker } from 'react-native-maps';

function BuildingIdentification () {
    return (
        <View>
            <Marker
                coordinate={{ latitude: 45.497222, longitude: -73.578809 }}>
                <Text style={styles.buildingIdentification}>H</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.496537, longitude: -73.577755 }}>
                <Text style={styles.buildingIdentification}>LB</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.495804, longitude: -73.578764 }}>
                <Text style={styles.buildingIdentification}>GM</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.495431, longitude: -73.578133 }}>
                <Text style={styles.buildingIdentification}>EV</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.495306, longitude: -73.579011 }}>
                <Text style={styles.buildingIdentification}>MB</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.457825, longitude: -73.641458 }}>
                <Text style={styles.buildingIdentification}>SP</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.457516, longitude: -73.640327 }}>
                <Text style={styles.buildingIdentification}>CJ</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.458094, longitude: -73.639768 }}>
                <Text style={styles.buildingIdentification}>AD</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.458314, longitude: -73.640409 }}>
                <Text style={styles.buildingIdentification}>CC</Text>
            </Marker>

            <Marker
                coordinate={{ latitude: 45.493450, longitude: -73.576814 }}>
                <Text style={styles.buildingIdentification}>GN</Text>
            </Marker>
        </View>
    );
}

export const styles = StyleSheet.create({
    buildingIdentification: {
        color: "white",
        fontSize: 30,
        fontWeight: "bold",
        textShadowColor: '#000000',
        textShadowOffset: { width: 1, height: 4 },
        textShadowRadius: 5
    }
})

export { BuildingIdentification };