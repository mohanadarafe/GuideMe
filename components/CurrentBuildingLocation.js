import React, { useEffect } from "react";
import coord from "../constants/buildingCoordinates";
import { isPointInPolygon } from "geolib";
import { AsyncStorage, StyleSheet, View, Text, Button, TouchableOpacity  } from "react-native";
import * as Location from "expo-location";
import Modal from "react-native-modal";
import MaterialIcons from "react-native-vector-icons/MaterialCommunityIcons";


/**
 * US5 - As a user, I would like to know which building I am currently in
 * US32 - As a user, I would like to be able to know where I am indoors.
 * The following function provides the building & current floor of a user.
 * 
 * Note: call CurrentBuildingLocation() inside BottomMenu.js
 */
function CurrentBuildingLocation () {
    const [currentBuilding, setCurrentBuilding] = React.useState("");
    const [lastLat, setlastLat] = React.useState(0);
    const [lastLong, setlastLong] = React.useState(0);
    const [altitude, setAltitude] = React.useState("0");
    const [modalVisibility, setModalVisibility] = React.useState(false);

    // These are not the real values. We must measure these values real time...
    // let baseAltitude = 35;
    // let floorHeight = 5;
    // let currentFloor = 0;

    const _getLocationAsync = async () => {
               let location = await Location.getCurrentPositionAsync({/*insert acuracy*/ });
        setlastLat(JSON.stringify(location.coords.latitude));
        setlastLong(JSON.stringify(location.coords.longitude));
        setAltitude(JSON.stringify(location.coords.altitude));
    };

    AsyncStorage.setItem("altitude", altitude);
    AsyncStorage.setItem("currentBuilding", currentBuilding);
    

    useEffect(() => {
        const intervalId = setInterval(() => {

            _getLocationAsync();

            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.gn.coordinates[0].latitude, longitude: coord.gn.coordinates[0].longitude },
                { latitude: coord.gn.coordinates[1].latitude, longitude: coord.gn.coordinates[1].longitude },
                { latitude: coord.gn.coordinates[2].latitude, longitude: coord.gn.coordinates[2].longitude },
                { latitude: coord.gn.coordinates[3].latitude, longitude: coord.gn.coordinates[3].longitude },
                { latitude: coord.gn.coordinates[4].latitude, longitude: coord.gn.coordinates[4].longitude }])) {

                setCurrentBuilding(coord.gn.name);


            }

            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.h.coordinates[0].latitude, longitude: coord.h.coordinates[0].longitude },
                { latitude: coord.h.coordinates[1].latitude, longitude: coord.h.coordinates[1].longitude },
                { latitude: coord.h.coordinates[2].latitude, longitude: coord.h.coordinates[2].longitude },
                { latitude: coord.h.coordinates[3].latitude, longitude: coord.h.coordinates[3].longitude },
                { latitude: coord.h.coordinates[4].latitude, longitude: coord.h.coordinates[4].longitude }])) {

                setCurrentBuilding(coord.h.name);

            }

            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.mb.coordinates[0].latitude, longitude: coord.mb.coordinates[0].longitude },
                { latitude: coord.mb.coordinates[1].latitude, longitude: coord.mb.coordinates[1].longitude },
                { latitude: coord.mb.coordinates[2].latitude, longitude: coord.mb.coordinates[2].longitude },
                { latitude: coord.mb.coordinates[3].latitude, longitude: coord.mb.coordinates[3].longitude },
                { latitude: coord.mb.coordinates[4].latitude, longitude: coord.mb.coordinates[4].longitude }])) {

                setCurrentBuilding(coord.mb.name);

            }

            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.ev.coordinates[0].latitude, longitude: coord.ev.coordinates[0].longitude },
                { latitude: coord.ev.coordinates[1].latitude, longitude: coord.ev.coordinates[1].longitude },
                { latitude: coord.ev.coordinates[2].latitude, longitude: coord.ev.coordinates[2].longitude },
                { latitude: coord.ev.coordinates[3].latitude, longitude: coord.ev.coordinates[3].longitude },
                { latitude: coord.ev.coordinates[4].latitude, longitude: coord.ev.coordinates[4].longitude }])) {

                setCurrentBuilding(coord.ev.name);

            }

            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.lb.coordinates[0].latitude, longitude: coord.lb.coordinates[0].longitude },
                { latitude: coord.lb.coordinates[1].latitude, longitude: coord.lb.coordinates[1].longitude },
                { latitude: coord.lb.coordinates[2].latitude, longitude: coord.lb.coordinates[2].longitude },
                { latitude: coord.lb.coordinates[3].latitude, longitude: coord.lb.coordinates[3].longitude },
                { latitude: coord.lb.coordinates[4].latitude, longitude: coord.lb.coordinates[4].longitude }])) {

                setCurrentBuilding(coord.lb.name);

            }
        }, 1000);
        return () => clearInterval(intervalId);
    });

    //Compute the floor level
    // currentFloor = (altitude - baseAltitude) / floorHeight;
    // roundedCurrentFloor = "Floor level: " + (Math.round(currentFloor * 100) / 100).toFixed(0);


    return (
        <TouchableOpacity style ={styles.layout}>
        <View>
            <MaterialIcons name="office-building" size={35} color="white" onPress={() => setModalVisibility(true)}></MaterialIcons><Modal isVisible={modalVisibility}>
               {currentBuilding === "" &&
                  <View style={styles.modal}>
                    <Text style={styles.modalText}>You have to be in a concordia building to use this feature</Text>
                    <Button style={styles.modalButton} title="Close" onPress={() => setModalVisibility(false)}/>
                  </View>
               }
               {currentBuilding !== "" &&
                   <View style={styles.modal}>
                       <Text style={styles.modalText}>{/*currentBuilding*/}We didn't measure the floor height yet... Waiting for concordia to reopen</Text>
                       <Text style={styles.modalText}>{/*roundedCurrentFloor*/}</Text>
                       <Button style={styles.modalButton} title="Close" onPress={() => setModalVisibility(false)}/>
                   </View>
               }
           </Modal>
        </View>
        </TouchableOpacity>
    );
}
export const styles = StyleSheet.create({
    layout: {
        width: 60,
        height: 60,
        borderRadius: 100/2,
        backgroundColor: "#2A2E43",
        justifyContent: "center",
        alignItems: "center"        //Align text horizontally
    },
    modal: {
        position:"absolute",
        right: "20%",
        paddingTop:10,
        borderRadius:10,
        width:200,
        backgroundColor: "white",
    },
    modalText: {
        fontSize:20,
        textAlign: 'center'         //Align text vertically
    },
});

export { CurrentBuildingLocation };