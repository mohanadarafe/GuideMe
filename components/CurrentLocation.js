import React, { useEffect, useState } from 'react';
import coord from '../constants/buildingCoordinates';
import { isPointInPolygon } from 'geolib'
import {AsyncStorage} from 'react-native';

export function CurrentLocation() {
    const [currentBuilding, setcurrentBuilding] = React.useState("")
    const [lastLat, setlastLat] = React.useState(0)
    const [lastLong, setlastLong] = React.useState(0)
    const [altitude, setAltitude] = React.useState("0") 

    let baseAltitude = 35;
    let floorHeight = 5;
    let currentFloor = 0;

    useEffect(() => {
        const intervalId = setInterval(() => {
            navigator.geolocation.getCurrentPosition((position) => {
                setlastLat(position.coords.latitude);
                setlastLong(position.coords.longitude);
                setAltitude(position.coords.altitude.toString());
            });
            
            AsyncStorage.setItem("altitude", altitude);
            AsyncStorage.setItem("currentBuilding", currentBuilding);

            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.gn.coordinates[0].latitude, longitude: coord.gn.coordinates[0].longitude },
                { latitude: coord.gn.coordinates[1].latitude, longitude: coord.gn.coordinates[1].longitude },
                { latitude: coord.gn.coordinates[2].latitude, longitude: coord.gn.coordinates[2].longitude },
                { latitude: coord.gn.coordinates[3].latitude, longitude: coord.gn.coordinates[3].longitude },
                { latitude: coord.gn.coordinates[4].latitude, longitude: coord.gn.coordinates[4].longitude }])) {
    
                setcurrentBuilding(coord.gn.name)
    
            }
    
            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.h.coordinates[0].latitude, longitude: coord.h.coordinates[0].longitude },
                { latitude: coord.h.coordinates[1].latitude, longitude: coord.h.coordinates[1].longitude },
                { latitude: coord.h.coordinates[2].latitude, longitude: coord.h.coordinates[2].longitude },
                { latitude: coord.h.coordinates[3].latitude, longitude: coord.h.coordinates[3].longitude },
                { latitude: coord.h.coordinates[4].latitude, longitude: coord.h.coordinates[4].longitude }])) {
    
                setcurrentBuilding(coord.h.name)
    
            }
    
            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.mb.coordinates[0].latitude, longitude: coord.mb.coordinates[0].longitude },
                { latitude: coord.mb.coordinates[1].latitude, longitude: coord.mb.coordinates[1].longitude },
                { latitude: coord.mb.coordinates[2].latitude, longitude: coord.mb.coordinates[2].longitude },
                { latitude: coord.mb.coordinates[3].latitude, longitude: coord.mb.coordinates[3].longitude },
                { latitude: coord.mb.coordinates[4].latitude, longitude: coord.mb.coordinates[4].longitude }])) {
    
                setcurrentBuilding(coord.mb.name)
    
            }
    
            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.ev.coordinates[0].latitude, longitude: coord.ev.coordinates[0].longitude },
                { latitude: coord.ev.coordinates[1].latitude, longitude: coord.ev.coordinates[1].longitude },
                { latitude: coord.ev.coordinates[2].latitude, longitude: coord.ev.coordinates[2].longitude },
                { latitude: coord.ev.coordinates[3].latitude, longitude: coord.ev.coordinates[3].longitude },
                { latitude: coord.ev.coordinates[4].latitude, longitude: coord.ev.coordinates[4].longitude }])) {
    
                setcurrentBuilding(coord.ev.name)
    
            }
    
            if (isPointInPolygon({ latitude: lastLat, longitude: lastLong },
                [{ latitude: coord.lb.coordinates[0].latitude, longitude: coord.lb.coordinates[0].longitude },
                { latitude: coord.lb.coordinates[1].latitude, longitude: coord.lb.coordinates[1].longitude },
                { latitude: coord.lb.coordinates[2].latitude, longitude: coord.lb.coordinates[2].longitude },
                { latitude: coord.lb.coordinates[3].latitude, longitude: coord.lb.coordinates[3].longitude },
                { latitude: coord.lb.coordinates[4].latitude, longitude: coord.lb.coordinates[4].longitude }])) {
    
                setcurrentBuilding(coord.lb.name)
    
            }

        }, 1000)
        return () => clearInterval(intervalId);
    })

    currentFloor = (altitude - baseAltitude)/floorHeight;

    if(currentFloor > 0){
        roundedCurrentFloor = "This is the " + (Math.round(currentFloor * 100) / 100).toFixed(0) + "rd floor";
    }

    if (currentBuilding === "") {
        return ("");
    }
    return (currentBuilding + " - " + roundedCurrentFloor);

}