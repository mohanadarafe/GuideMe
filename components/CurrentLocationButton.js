import React from "react";
import MaterialIcons from "react-native-vector-icons/MaterialCommunityIcons";
import { StyleSheet, TouchableOpacity } from "react-native";
import { View } from "native-base";

function CurrentLocationButton ({mapReference}) {

    const goToCurrentLocation = () => { navigator.geolocation.getCurrentPosition(
        ({ coords }) => {
          if (mapReference) {
            mapReference.current.animateToRegion({
              latitude: coords.latitude,
              longitude: coords.longitude,
              latitudeDelta: 0.005,
              longitudeDelta: 0.005
            })
          }
        },
        (error) => alert('Error: Are location services on?'),
        { enableHighAccuracy: true }
      )
    }

    return (
        <TouchableOpacity style ={styles.layout} onPress = {goToCurrentLocation}>
        <View style ={{top: "2%"}}>
            <MaterialIcons name="target" size={35} color="white"></MaterialIcons>
        </View>
        </TouchableOpacity>
    );
}

export const styles = StyleSheet.create({
    layout: {
        width: 60,
        height: 60,
        borderRadius: 100/2,
        backgroundColor: "#2A2E43",
        justifyContent: "center",
        alignItems: "center"
    }
});

export default CurrentLocationButton;