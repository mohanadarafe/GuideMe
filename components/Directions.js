/* eslint-disable no-unused-vars */
import React, { useRef } from "react";
import { StyleSheet, TouchableOpacity } from "react-native";
import MapView, { PROVIDER_GOOGLE } from "react-native-maps";
import { View, Text, Icon } from "native-base";
import PropTypes from "prop-types";


const mapPosition = {
    sgwCoord: {
        latitude: 45.496557,
        longitude: -73.578896,
        latitudeDelta: 0.005,
        longitudeDelta: 0.005,
    },
    loyCoord: {
        latitude: 45.457841,
        longitude: -73.640307,
        latitudeDelta: 0.01,
        longitudeDelta: 0.01,
    }
};

/**		
* US1 - As a user, I would like to navigate through SGW campus.		
* US2 - As a user, I would like to navigate through Loyola campus.		
* 		
* This is our main screen which includes all the components inside a map.		
*/
function Directions(props) {

    const [decodedPolylines, setDecodedPolylines] = React.useState([]);
    const [detailedInstructions, setDetailedInstructions] = React.useState();
    const [directionRegion, setDirectionRegion] = React.useState(props.initialRegion);
    const mapRef = useRef(null);

    const handleMapPress = () => {
        const region = {
            latitude: 37.78825,
            longitude: -122.4324,
            latitudeDelta: 0.0922,
            longitudeDelta: 0.0421
        };
        mapRef.current.animateToRegion(region, 1000);
    };

    const goBackPressHandler = () => {
        props.navigation.goBack();
    };

    // useEffect(() => {		
    //     const fetchData = async () =>  {		
    //         try{		
    //             let resp = await fetch(`https://maps.googleapis.com/maps/api/directions/json?origin=${props.origin}&destination=${props.destination}&key=AIzaSyC_ik7PAKgcFPtFYnDAqCr3TI7HM9QU_SY`);		
    //             const jsonResponse = await resp.json();		
    //             const points = PolyLine.decode(jsonResponse.routes[0].overview_polyline.points);		
    //             const decodedPoints = points.map(point => {		
    //                 return {		
    //                     latitude: point[0],		
    //                     longitude: point[1]		
    //                 }		
    //             });		
    //             setDecodedPolylines(decodedPoints);		
    //             // const stepPoint = PolyLine.decode(jsonResponse.routes[0].)		
    //             setDirectionRegion()		
    //             setDetailedInstructions(jsonResponse.routes[0].legs);		
    //         } catch(error) {		
    //             console.log(error);		
    //         }		
    //     }		
    //     fetchData();		
    // }, []);		

    const onLayout = () => {
        setTimeout(() => {
            mapRef.fitToCoordinates([
                { latitude: 33.9260206, longitude: 75.0173499 }, { latitude: 33.949817, longitude: 74.985655 }],
                { edgePadding: { bottom: 300, right: 0, left: 0, top: 0 }, animated: true, });
        }, 2000);
    };
    return (
        <View>
            <MapView
                ref={mapRef}
                style={styles.map}
                provider={PROVIDER_GOOGLE}
                region={directionRegion}
                showsUserLocation={true}
                showsCompass={true}
                showsBuildings={true}
                onPress={onLayout}
            >
                {/* <Polyline		
                 coordinates = {decodedPolylines}		
                 strokeWidth = {2}		
                 strokeColor = "pink"		
                 /> */}
            </MapView>

            <View style={styles.navigationHeader}>
                <View style={{ top: "25%" }}>
                    <TouchableOpacity onPress={goBackPressHandler}>
                        <Icon name="md-arrow-round-back" style={styles.backIcon}></Icon>
                    </TouchableOpacity>

                    <View style={styles.directionText}>
                        <Text style={styles.DirectionTextHeader}>Route Directions</Text>
                        <View style={styles.lineHeader}></View>
                    </View>

                </View>
            </View>
            <View style={styles.bottomArrowDirectionContainer}>
                <TouchableOpacity style={styles.arrowDirection}>
                    <View>
                        <Icon name="arrow-back"></Icon>
                    </View>
                </TouchableOpacity>
                <TouchableOpacity style={styles.arrowDirection}>
                    <View >
                        <Icon name="arrow-forward"></Icon>
                    </View>
                </TouchableOpacity>
            </View>
            {/* <BottomMenu/> */}
        </View>
    );
}

Directions.propTypes = {
    navigation: PropTypes.any,
    goBack: PropTypes.func,
    initialRegion: PropTypes.any
};

export const styles = StyleSheet.create({
    map: {
        height: "100%"
    },
    navigationHeader: {
        width: "100%",
        height: "25%",
        flexDirection: "column",
        backgroundColor: "#2A2E43",
        position: "absolute"
    },
    directionText: {
        justifyContent: "center",
        alignItems: "center"
    },
    DirectionTextHeader: {
        color: "white",
        fontSize: 25
    },
    lineHeader: {
        borderBottomColor: "white",
        width: "100%",
        borderBottomWidth: 2,
        top: "10%"
    },
    backIcon: {
        color: "white",
        left: "5%"
    },
    bottomArrowDirectionContainer: {
        top: "90%",
        left: "70%",
        position: "absolute",
        height: 50,
        width: 100,
        flexDirection: "row",
        justifyContent: "space-between"
    },
    arrowDirection: {
        width: "45%",
        color: "#2A2E43",
        backgroundColor: "white",
        alignItems: "center",
        justifyContent: "center",
        borderWidth: 2,
        borderColor: "#2A2E43",
        borderRadius: 5
    }
});

export default Directions;