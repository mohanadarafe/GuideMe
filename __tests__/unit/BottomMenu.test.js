import React from "react";
import { BottomMenu } from '../../components/BottomMenu';
import renderer from 'react-test-renderer';

describe("Bottom Menu component", () => {
    test('renders correctly', () => {
        const tree = renderer.create(<BottomMenu />).toJSON();
        expect(tree).toMatchSnapshot();
      });
});
