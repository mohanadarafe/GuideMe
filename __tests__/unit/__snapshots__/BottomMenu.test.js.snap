// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom Menu component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2A2E43",
      "borderRadius": 30.5,
      "bottom": -275,
      "height": 350,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "encodeSansExpanded",
        "fontSize": 20,
        "left": "12.5%",
        "position": "absolute",
        "top": "5%",
      }
    }
  >
    Nearby
  </Text>
  <Text
    style={
      Object {
        "color": "#80828D",
        "fontFamily": "encodeSansExpanded",
        "fontSize": 16,
        "left": "12.5%",
        "position": "absolute",
        "top": "12%",
      }
    }
  >
    Food, drinks & more
  </Text>
  <View
    style={
      Object {
        "left": "80%",
        "position": "absolute",
        "top": "6.5%",
      }
    }
  >
    <RCTSwitch
      accessibilityRole="button"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="green"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      tintColor="red"
      value={true}
    />
  </View>
</View>
`;
